subxt/src/rpc/lightclient/background.rs

```diff
@@ -0,0 +1,275 @@
+                // Drain the subscription cache, that holds messages that
+                // were not propagated to this subscription yet (because the
+                // RPC server produced a notification before the user registered
+                // to receive notifications).
+                if let Some(cached_responses) = self.subscriptions_cache.remove(&id) {
+                    tracing::debug!(target: LOG_TARGET, "Some messages were cached before susbcribing");
+
+                    for response in cached_responses {
+                        if sender.send(response).await.is_err() {
+                            tracing::warn!(target: LOG_TARGET, "Cannot send notification to susbcription {:?}", id);
+                        }
+                    }
+                }
+
+                self.subscriptions.insert(id, sender);
+            }
+        };
+    }
+
+    /// Parse the response received from the light client and sent it to the appropriate user.
+    async fn handle_rpc_response(&mut self, response: String) {
+        match RpcResponse::from_str(&response) {
+            Ok(RpcResponse::Method { id, result }) => {
+                // Send the response back.
+                if let Some(sender) = self.requests.remove(&id) {
+                    if sender.send(result).is_err() {
+                        tracing::warn!(target: LOG_TARGET, " Cannot send method response to id {:?}", id);
+                    }
+                }
+            }
+            Ok(RpcResponse::Subscription { method, id, result }) => {
+                // Subxt calls into `author_submitAndWatchExtrinsic`, however the smoldot produces
+                // `{"event":"broadcasted","numPeers":1}` which is part of the RPC V2 API. Ignore
+                // this spurious event.
+                if method == "transaction_unstable_watchEvent"
+                    && result.to_string().contains("broadcasted")
+                {
+                    tracing::debug!(target: LOG_TARGET, "Ignoring notification {:?}", result);
+                    return;
+                }
+
+                if let Some(sender) = self.subscriptions.get_mut(&id) {
+                    // Send the current notification response.
+                    if sender.send(result).await.is_err() {
+                        tracing::warn!(target: LOG_TARGET, "Cannot send notification to susbcription {:?}", id);
+                    }
+                    return;
+                }
+
+                // Subscription ID not registered yet, cache the response.
+                // Note: Compiler complains about moving the `result` for `.entry.and_modify(..).or_insert(..)`,
+                // because it sees it as used on both closures and apparently cannot determine that only one
+                // closure can be executed.
+                match self.subscriptions_cache.entry(id) {
+                    std::collections::hash_map::Entry::Occupied(mut entry) => {
```

Comment:
nit, import `std::collections::hash_map::Entry` to avoid repeating here and a few lines below?